mono/metadata/mono-perfcounters.c

```diff
@@ -1693,40 +1730,48 @@ get_string_array (void **array, int count, gboolean is_process, MonoError *error
 			sprintf (buf, "%d", GPOINTER_TO_INT (array [i]));
 			p = buf;
 		}
-		MonoString *str = mono_string_new_checked (domain, p, error);
+		MonoStringHandle str;
+		SETUP_ICALL_FRAME;
+		str = mono_string_new_utf8_len (domain, p, strlen (p), error);
 		if (p != buf)
 			g_free (p);
-		return_val_if_nok (error, NULL);
-		mono_array_setref_internal (res, i, str);
+		MONO_HANDLE_ARRAY_SETREF (res, i, str);
```

Comment:
Previous code skipped the array_setref in the error path.

Reply:
But the array is probably dead/unseen in that case, just garbage, in which case ok. Same with next chunk.